lib/spack/spack/platforms/_platform.py

```diff
@@ -45,82 +25,72 @@ class attributes such as priority, front_target, back_target, front_os, back_os.
     #: binary formats used on this platform; used by relocation logic
     binary_formats = ["elf"]
 
-    front_end: Optional[str] = None
-    back_end: Optional[str] = None
-    default: Optional[str] = None  # The default back end target.
-
-    front_os: Optional[str] = None
-    back_os: Optional[str] = None
+    default: Optional[str] = None
     default_os: Optional[str] = None
```

Comment:
* `default_os` can be named `os` and have type `str` (not sure why it's set `None` on the class level?)
* `default` can be named `target` with type `str`?

Reply:
Changed the types, for naming I would defer to another PR for a few reasons:
- There is currently a `Platform.target` method, so target isn't a good option
- The default target is just the default, other targets can be selected though (build for `x86_64_v3`, while the default is `icelake`)
- The test platform has two operating systems, changing that probably means fixing a few tests